docs/fundamentals/code-analysis/configuration-files.md

```diff
@@ -0,0 +1,135 @@
+
+You can apply EditorConfig file conventions to a folder, a project, or an entire repo by placing the file in the corresponding directory. These options will be applied when executing the analysis at build time as well as inside Visual Studio.
+
+If you have an existing *.editorconfig* file for editor settings such as indent size or whether to trim trailing whitespace, you can place your code analysis configuration options in the same file.
+
+> [!TIP]
+> Visual Studio provides an *.editorconfig* item template that makes is easy to add one of these files to your project. For more information, see [Add an EditorConfig file to a project](/visualstudio/ide/create-portable-custom-editor-options#add-an-editorconfig-file-to-a-project).
+
+### Example
+Following is an example EditorConfig file to configure options and rule severity:
+
+```ini
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
+
+# C# files
+[*.cs]
+
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+#### .NET Coding Conventions ####
+
+# this. and Me. preferences
+dotnet_style_qualification_for_method = true:warning
+
+#### Diagnostic configuration ####
+
+# CA1000: Do not declare static members on generic types
+dotnet_diagnostic.CA1000.severity = warning
+```
+
+## Global AnalyzerConfig
+
+Starting with the .NET 5.0 SDK (which is supported starting with Visual Studio 2019 version 16.8), you can also configure analyzer options with global AnalyzerConfig files. These files are used to provide **options that apply to all the source files in a project**, regardless of their file names or file paths.
+
+Unlike [EditorConfig](#editorconfig) files, global config files can't be used to configure editor style settings for IDEs, such as indent size or whether to trim trailing whitespace. Instead, they are designed purely for specifying project-level analyzer configuration options.
+
+### Format
+
+Unlike EditorConfig files, which must have section headers, such as `[*.cs]`, to identify the applicable files and folders, global AnalyzerConfig files don't have section headers. Instead, they require a top level entry of the form `is_global = true`. This indicates that all the options in the file apply globally to the entire project. For example:
+
+```ini
+is_global = true
+<option_name> = <option_value>
+```
+
+### Naming
+
+Unlike EditorConfig files, which must be named `.editorconfig`, global config files do not need to have a specific name or extension. However, if you name these files as `.globalconfig` then they will be implicitly applied to all the C# and Visual Basic projects within the current folder, including subfolders. Otherwise, you will need to explicitly add the following MSBuild item group to your MSBuild project file:
```

Comment:
```suggestion
Unlike EditorConfig files, which must be named `.editorconfig`, global config files do not need to have a specific name or extension. However, if you name these files as `.globalconfig` then they will be implicitly applied to all the C# and Visual Basic projects within the current folder, including subfolders. Otherwise, you must explicitly add the `GlobalAnalyzerConfigFiles` item to your MSBuild project file:
```